src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -410,6 +420,30 @@ type DiagnosticItem =
         with
         | _ -> str // let's fail silently for now
 
+    // TODO: When the names of the runtime capabilities are finalized, they can be included in the result
+    //       comparison error messages.
+    static member private CapabilityErrorFormatString(code) =
+        match code with
+        | ErrorCode.UnsupportedResultComparison -> "The target {0} does not support comparing measurement results."
```

Comment:
(nit)
At certain stage of my career I came to an impression that it is not always a good idea to provide a dot (full stop) `.` in the end of a message, because some subsequent code may want to surround the message with quotes `"` or single quotes `'` or parentheses `()`, etc. E.g. the subsequent code may want to print:
> The process failed (The target MyTestTarget does not support comparing measurement results.). 

In the fragment above the presence of a dot `.` in front of the closing parenthesis `)` is not what some users would expect to see.

Consider not providing the dot `.` in the end of the messages (but in the middle of the multi-sentence messages seems reasonable).

Reply:
Interesting observation. However, there're about other 500 error messages in this file and all of them have period in the end. I'd say it's better to be uniform in this case.